app/uk/gov/hmrc/exports/services/DeclarationService.scala

```diff
@@ -76,6 +77,16 @@ class DeclarationService @Inject() (declarationRepository: DeclarationRepository
   def find(search: DeclarationSearch, pagination: Page, sort: DeclarationSort): Future[Paginated[ExportsDeclaration]] =
     declarationRepository.find(search, pagination, sort)
 
+  def findDraftByParent(eori: Eori, parentId: String): Future[Option[ExportsDeclaration]] = {
+    val filter = Json.obj(
+      "eori" -> eori,
+      "declarationMeta.parentDeclarationId" -> parentId,
+      "declarationMeta.status" -> Json.obj("$in" -> Json.arr(AMENDMENT_DRAFT.toString, DRAFT.toString)),
+      "declarationMeta.parentDeclarationEnhancedStatus" -> EnhancedStatus.ERRORS.toString
```

Comment:
Can we actually have declarations not submitted yet (accordingly in **DRAFT**) with "ERROR" EnhancedStatus?

Reply:
It's the draft declaration's parent status, not the draft's status itself.